index.js

```diff
@@ -12,3 +15,27 @@ const pagination = document.querySelector('[data-js="pagination"]');
 const maxPage = 1;
 const page = 1;
 const searchQuery = "";
+
+// Fetching Data
+async function fetchCharacters(slug) {
```

Comment:
I recommend to not only fetch the data in this function but to also process the data here directly. Meaning, after fetching the data, you can directly use the data to create the character cards and append all cards to the `cardContainer`. As you only need to create the cards after fetching the data, it makes sense to include the card creating process inside `fetchCharacters`.